test/vfd_swmr_bigset_writer.c

```diff
@@ -1097,89 +1841,168 @@ main(int argc, char **argv)
         }
 
         /* config, tick_len, max_lag, writer, flush_raw_data, md_pages_reserved, md_file_path */
-        init_vfd_swmr_config(&config, 4, 7, s.writer, FALSE, 128, "./bigset-shadow-%zu", i);
+        init_vfd_swmr_config(&config, TICK_LEN, MAX_LAG, s.writer, FALSE, 128, "./bigset-shadow-%zu", i);
 
         /* use_latest_format, use_vfd_swmr, only_meta_page, config */
-        fapl = vfd_swmr_create_fapl(true, s.use_vfd_swmr, true, &config);
-
-        if (fapl < 0)
-            errx(EXIT_FAILURE, "vfd_swmr_create_fapl");
+        if ((fapl = vfd_swmr_create_fapl(true, s.use_vfd_swmr, true, &config)) < 0) {
+            fprintf(stderr, "vfd_swmr_create_fapl failed\n");
+            TEST_ERROR;
+        }
 
         s.file[i] = s.writer ? H5Fcreate(s.filename[i], H5F_ACC_TRUNC, fcpl, fapl)
                              : H5Fopen(s.filename[i], H5F_ACC_RDONLY, fapl);
 
-        if (s.file[i] == badhid)
-            errx(EXIT_FAILURE, s.writer ? "H5Fcreate" : "H5Fopen");
+        if (s.file[i] == badhid) {
+            fprintf(stderr, s.writer ? "H5Fcreate failed" : "H5Fopen failed");
+            TEST_ERROR;
+        }
 
-        if (H5Pclose(fapl) < 0)
-            errx(EXIT_FAILURE, "H5Pclose(fapl)");
+        if (H5Pclose(fapl) < 0) {
+            fprintf(stderr, "H5Pclose failed\n");
+            TEST_ERROR;
+        }
     }
 
-    block_signals(&oldsigs);
+    /* Initiailze named pipes */
+    if (s.use_named_pipe && !np_init(&np, s.writer)) {
+        fprintf(stderr, "np_init() failed\n");
+        TEST_ERROR;
+    }
 
     if (s.writer) {
+        /* Writer tells reader to start */
+        np.notify = 1;
+        if (s.use_named_pipe && HDwrite(np.fd_writer_to_reader, &np.notify, sizeof(int)) < 0) {
+            fprintf(stderr, "HDwrite failed\n");
+            TEST_ERROR;
+        }
+
+        /* Create NDATASETS datasets as the reader is doing verification.  So no communication with
+         * the reader during the creation of datasets.
+         */
         for (which = 0; which < s.ndatasets; which++)
-            create_extensible_dset(&s, which);
+            if (!create_extensible_dset(&s, which)) {
+                fprintf(stderr, "create_extensible_dset failed: number %u\n", which);
+                TEST_ERROR;
+            }
 
-        for (step = 0; step < s.nsteps; step++) {
-            for (which = 0; which < s.ndatasets; which++) {
-                dbgf(2, "step %d which %d\n", step, which);
-                write_extensible_dset(&s, which, step, mat);
-                if (s.ndatasets <= s.nsteps)
-                    nanosleep(&s.update_interval, NULL);
+        /* Call H5Fvfd_swmr_end_tick to end the tick.  No communication with the reader in this step */
```

Comment:
Just to educate me, why do we call "H5Fvfd_swmr_end_tick" here?

Reply:
To make sure the metadata is visible to the reader.